### DISCUSSION TITLE
有没有考虑用msw来做拦截？

### DISCUSSION THREAD

**Discussion by liufeisprit** [PengChen96/ajax-tools#19] [Ideas]
目前用的重写原生的xhr还是太暴力了 侵入性挺大的。 mock的数据在浏览器devtools是看不到的，如果用service worker可以完美解决这个问题。如果有这个意愿 我可以参与贡献

**Comment by PengChen96** [comment]
可以简单说下方案吗，有更好的方式，非常欢迎参与哈。

**Reply by liufeisprit** [reply]
https://mswjs.io/docs 可以看下msw的文档 是一个很成熟的mock库  基于Service Worker对请求响应来做拦截的。 对主流程不会有影响 对响应的内容并不是简单的替换。 这里可以看下一个demo  https://tiven.cn/service/demos/msw-tools
而且也可以同样满足现在这个插件的功能 自定义响应请求等。 可以考虑一下是否可行 🤔

**Reply by PengChen96** [reply]
简单看了下还不错，感觉可以尝试一下。

**Comment by liufeisprit** [comment]
深入看了下 chrome插件应该不能向页面插入web的service-worker 注册service-worker的文件地址要是web主域名下的
